### PR TITLE
Tabbing NewMessage operation, making whatsapp required for MTM

### PR DESCRIPTION
**Pull request by slorello89**
# Description

Refreshing the Olympus OAS spec so that it can be a bit more usable then it is now. Making all channel/messageType permutations tabbable. Also made `whatsapp` node a required part of a WhatsApp message for MTM messages.

Example:

# New 

* Tab interface for `NewMessage` operation
* Made `whatsapp` node a required field for MTM
* Cleaned up descriptions so they only refer to the channel for the tab

# Fixes

* Minor fixes to descriptions.

# Checklist

- [x] version number incremented (in the `info` section of the spec)

### CODE REVIEW

**Comment by mheap** [comment]
@slorello89 It looks like `Inbound Message` also needs updating

Also, if you provide the same `x-tab-id` for `Image`, `Template` etc it will automatically switch all subtabs when you select that value. So you can select FB->Image, and click WA and it'll have Image selected. Then if you click `Text` and go back to FB it'll have `Text` selected

**Comment by dragonmantank** [comment]
Is there a way to re-deploy this for testing inside NDP/station?

**Comment by lornajane** [comment]
I recreated the review app by removing the `review-app` label and then re-adding it (it helps to pause in the middle so GitHub gets both actions), does that help?

**Comment by dragonmantank** [comment]
> I recreated the review app by removing the `review-app` label and then re-adding it (it helps to pause in the middle so GitHub gets both actions), does that help?

That does, thank you!

File: definitions/messages-olympus.yml [hunk 1]
```diff
@@ -68,12 +68,12 @@ paths:
                 content:
                   application/json:
                     schema:
-                      $ref: "#/components/schemas/InboundMessage"
+                      $ref: "#/components/schemas/inbound-message"
```
Comment (dragonmantank) [inline definitions/messages-olympus.yml:71]: Why are we leaving this as hyphenated by camelCasing the others?

File: definitions/messages-olympus.yml [hunk 2]
```diff
@@ -68,12 +68,12 @@ paths:
...
               responses:
                 "200":
                   description: Your server returns this code if it accepts the callback.
 
-components:
+components:
```
Comment (dragonmantank) [inline definitions/messages-olympus.yml:76]: extra whitespace

File: definitions/messages-olympus.yml [hunk 3]
```diff
@@ -82,25 +82,1186 @@ components:
     basicAuth:
       type: http
       scheme: basic
-
   schemas:
-    NewMessage:
+    inbound-common:
```
Comment (dragonmantank) [inline definitions/messages-olympus.yml:86]: Same as above, but for all the names here. Most of the other APIs seem to use camelCase for schemas (though I noticed a few used PascalCase). For consistency we should at least stick to one for an API, and longer term clean up all of them to match.

**Comment by lornajane** [comment]
I'm confused about the status of this PR. It has two approvals but neither one merged the changes. Also does it have any relation to the other updates in https://github.com/Nexmo/api-specification/pull/300?

For naming things, we kind of have gone mostly CamelCase for things inside a spec but there is no official style guide. +1 for keeping consistency within a spec for now I think.

**Comment by slorello89** [comment]
> I'm confused about the status of this PR. It has two approvals but neither one merged the changes. Also does it have any relation to the other updates in #300?
> 
> For naming things, we kind of have gone mostly CamelCase for things inside a spec but there is no official style guide. +1 for keeping consistency within a spec for now I think.

Aye I agree - with consistency, at least within the spec - an oversight on my part.

**Comment by dragonmantank** [comment]
> It has two approvals but neither one merged the changes.

From my end I almost never merge someone else's PR. I leave that up to the person submitting the PR or the one "most responsible" for finishing out the lifecycle of a code change.

File: definitions/messages-olympus.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 openapi: "3.0.0"
 info:
-  version: 0.3.8
+  version: 0.3.9
```
Comment (tbedford) [inline definitions/messages-olympus.yml:3]: Possible race hazard with version number specified in #359

**Review by tbedford — APPROVED** [review]
Apart from my separate comment on version number, looks good.

**Pull request merged**